runner/internal/metrics/type.go

```diff
@@ -0,0 +1,49 @@
+package metrics
+
+import (
+	"reflect"
+	"time"
+)
+
+// Type represents the underlying type of a Value.
+type Type uint8
+
+const (
+	lastGoReflectKind = reflect.UnsafePointer
+
+	// TypeInvalid corresponds to an invalid type.
+	TypeInvalid = Type(reflect.Invalid)
+	// TypeInt corresponds to type int.
+	TypeInt = Type(reflect.Int)
+	// TypeDuration corresponds to type time.Duration.
+	TypeDuration = Type(lastGoReflectKind + iota)
+)
+
+// String returns a human-readable representation of the field.
+//
+// Example:
+//
+//	TypeDuration.String() == "time.Duration"
+//	Type(123).String() == "[unknown type]"
+func (typ Type) String() string {
```

Comment:
This method is currently not used, do we expect a usecase for it?

Reply:
It is used, although subtly (`String` is automatically called in string formatting function)
https://github.com/benchttp/engine/blob/a948c6c5ac99a8d6e7a2d43cef035594f6d618d1/internal/configparse/parse.go#L344-L349

Reply:
Oh yes totally!